### PR TITLE
add connection retry

### PR DESCRIPTION
**Pull request by matiascalegaris**
add connection dialog with timeout to retry connection with another ip

move power up time text from screen and add color when time is running out

Disable VSync option since is broken now and user don't understand why they have black screen

### CODE REVIEW

File: CODIGO/modNetwork.bas [hunk 1]
```diff
@@ -18,19 +18,48 @@ Attribute VB_Name = "modNetwork"
...
 
 Private Client As Network.Client
+
+Private Type t_FailedIp
+    IP As String
+    Port As String
+End Type
+
+Dim FailedIpList(10) As t_FailedIp
+Public FaiedListSize As Integer
```
Comment (morgolock) [inline CODIGO/modNetwork.bas:28]: `FailedListSize`

File: CODIGO/frmOpciones.frm [hunk 2]
```diff
@@ -1680,7 +1680,7 @@ scrVolume_Change_Err:
 End Sub
 
 Private Sub VSync_Click()
```
Comment (morgolock) [inline CODIGO/frmOpciones.frm:1682]: Suggestiion, get rid of `VSyncActivado` everywhere in the code. Delete `Private Sub VSync_Click()`

File: CODIGO/modNetwork.bas [hunk 3]
```diff
@@ -18,19 +18,48 @@ Attribute VB_Name = "modNetwork"
...
 #End If
 
+Public Function IsFailedString(ByVal IP As String, ByVal Port As String)
+    Dim i As Integer
+    For i = 0 To UBound(FailedIpList) - 1
+        If FailedIpList(i).IP = IP And FailedIpList(i).Port = Port Then
+            IsFailedString = True
+            Exit Function
+        End If
+    Next i
```
Comment (morgolock) [inline CODIGO/modNetwork.bas:43]: Missing  `IsFailedString = False` before leaving the function or at the beginning

File: CODIGO/modNetwork.bas [hunk 4]
```diff
@@ -18,19 +18,48 @@ Attribute VB_Name = "modNetwork"
...
+End Function
+
+Public Function AddFailedIp(ByVal IP As String, ByVal Port As String) As Boolean
+    FailedIpList(FaiedListSize).IP = IP
+    FailedIpList(FaiedListSize).Port = Port
+    FaiedListSize = FaiedListSize + 1
+    AddFailedIp = True
+    If FaiedListSize = UBound(servers_login_connections) Then
```
Comment (morgolock) [inline CODIGO/modNetwork.bas:51]: For clarity and readability, I would

Move this out of the function 
`If FaiedListSize = UBound(servers_login_connections) Then
        AddFailedIp = False
    End If`
    
to `RetryWithAnotherIp` and turn `AddFailedIp` into a `sub` not a `function`

File: CODIGO/modNetwork.bas [hunk 5]
```diff
@@ -60,6 +90,21 @@ Public Sub Send(ByVal Buffer As Network.Writer)
     
     Call Buffer.Clear
 End Sub
+
+Public Sub RetryWithAnotherIp()
+    Call Disconnect
+    If Not AddFailedIp(IPdelServidor, PuertoDelServidor) Then
```
Comment (morgolock) [inline CODIGO/modNetwork.bas:96]: this is a bit consufing

I would say we do

```
call AddFailedIp(IPdelServidor, PuertoDelServidor)
If FaiedListSize = UBound(servers_login_connections) Then 
Unload frmConnecting
Exit Sub`

**Review by morgolock — CHANGES_REQUESTED** [review]

File: CODIGO/modNetwork.bas (hunk 3 above)
Comment (matiascalegaris) [reply CODIGO/modNetwork.bas:43]: false is default value

**Review by morgolock — APPROVED** [review]

**Pull request merged**